src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/RazorSourceGeneratorTestsBase.cs

```diff
@@ -80,15 +92,86 @@ protected static async ValueTask<(GeneratorDriver, ImmutableArray<AdditionalText
 
     protected static GeneratorRunResult RunGenerator(Compilation compilation, ref GeneratorDriver driver, params Action<Diagnostic>[] expectedDiagnostics)
     {
-        driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out compilation, out _);
+        return RunGenerator(compilation, ref driver, out _, expectedDiagnostics);
+    }
+
+    protected static GeneratorRunResult RunGenerator(Compilation compilation, ref GeneratorDriver driver, out Compilation outputCompilation, params Action<Diagnostic>[] expectedDiagnostics)
+    {
+        driver = driver.RunGeneratorsAndUpdateCompilation(compilation, out outputCompilation, out _);
 
-        var actualDiagnostics = compilation.GetDiagnostics().Where(d => d.Severity != DiagnosticSeverity.Hidden);
+        var actualDiagnostics = outputCompilation.GetDiagnostics().Where(d => d.Severity != DiagnosticSeverity.Hidden);
         Assert.Collection(actualDiagnostics, expectedDiagnostics);
 
         var result = driver.GetRunResult();
         return result.Results.Single();
     }
 
+    protected static async Task<string> RenderRazorPageAsync(Compilation compilation, string name)
+    {
+        // Load the compiled DLL.
+        var assemblyLoadContext = new AssemblyLoadContext("Razor execution", isCollectible: true);
+        Assembly assembly;
+        using (var peStream = new MemoryStream())
+        {
+            var emitResult = compilation.Emit(peStream);
+            Assert.True(emitResult.Success, string.Join(Environment.NewLine, emitResult.Diagnostics));
+            peStream.Position = 0;
+            assembly = assemblyLoadContext.LoadFromStream(peStream);
+        }
```

Comment:
Given the way we've setup the `AssemblyLoadContext` I _believe_ that any dependencies required by this compiled DLL will be loaded from the default `AssemblyLoadContext`. Essentially it will get dependencies from what the project pulls in. 

Are we okay with that? Essentially as we move the dependencies in our project forward it will move forward the dependencies the test execution uses. 

I'm okay with that but want to make sure collectively we understand how this is all tied together. 

Also open to hearing I'm wrong about the way dependencies load by default here. I'm more used to cases where we've heavily plugged dependency loading to be highly restricted. So not 100% sure what the default policy is. :smile: